docs/kube-burner.md

```diff
@@ -0,0 +1,116 @@
+
+## Supported workloads
+
+Ripsaw's kube-burner integration supports the following workloads:
+
+- **cluster-density**: This workload is a cluster density focused test that creates a set of Deployments, Builds, Secret, Services and Routes across the cluster. This is a namespaced workload, meaning that kube-burner **will create as many namespaces with these objects as the configured job_iterations**. 
+Each iteration of this workload creates the following objects:
+  - 12 imagestreams
+  - 3 buidconfigs
+  - 6 builds
+  - 1 deployment with 2 pod replicas (sleep) mounting two secrets each. *deployment-2pod*
+  - 2 deployments with 1 pod replicas (sleep) mounting two secrets. *deployment-1pod*
+  - 3 services, one pointing to *deployment-2pod*, and other two pointing to *deployment-1pod*.
+  - 3 route. 1 pointing to the service *deployment-2pod* and other two pointing to *deployment-1pod*
+  - 20 secrets
+
+- **kubelet-density**: Creates a single namespace with a number of Deployments equal to **job_iterations**.
+Each iteration of this workload creates the following object:
+  - 1 pod. (sleep)
+
+- **kubelet-density-heavy**. Creates a **single namespace with a number of applications equals to job_iterations**. This application consists on two deployments (a postgresql database and a simple client that generates some CPU load) and a service that is used by the client to reach the database.
+Each iteration of this workload creates the following objects:
+  - 1 deployment holding a postgresql database
+  - 1 deployment holding a client application for the previous database
+  - 1 service pointing to the postgresl database
+
+The workload type is specified by the parameter `workload` from the `args` object of the configuration. Each workload supports several configuration parameters, detailed in the [configuration section](#configuration)
+
+## Configuration
+
+All kube-burner's workloads support the following parameters:
+
+- **workload**: Type of kube-burner workload. As mentioned before, allowed values are cluster-density, kubelet-density and kubelet-density-heavy
+- **default_index**: Elasticsearch index name. Defaults to __ripsaw-kube-burner__
+- **job_iterations**: How many iterations to execute of the specified kube-burner workload
+- **qps**: Limit object creation queries per second. Defaults to __5__
+- **burst**: Maximum burst for throttle. Defaults to __10__
+- **image**: Allows to use an alternative kube-burner container image. Defaults to `quay.io/cloud-bulldozer/kube-burner:latest`
+- **wait_when_finished**: Makes kube-burner to wait for all objects created to be ready/completed before index metrics and finishing the job. Defaults to __true__
+- **pod_wait**: Wait for all pods to be running before moving forward to the next job iteration. Defaults to __false__
+- **verify_objects**: Verify object count after running each job. Defaults to __true__
```

Comment:
What's the default timeout after which we report the job as failed?

Reply:
We don't have timeout at the moment. Shall we add it?
Maybe something using activeDeadlineSeconds could do the trick. What do you think?

```
$ oc explain job.spec.activeDeadlineSeconds
KIND:     Job
VERSION:  batch/v1

FIELD:    activeDeadlineSeconds <integer>

DESCRIPTION:
     Specifies the duration in seconds relative to the startTime that the job
     may be active before the system tries to terminate it; value must be
     positive integer
```

Reply:
I finally added the parameter job_timeout. Defaults to 3600

Reply:
We might want the timeout to be applicable just for the "verify_objects" action run at the end of the job as it might be tricky to calculate the the total time a job might take from start to finish. Thoughts?

Reply:
Thinking a bit, we might also want to take the status of the objects into account i.e fail if the state of the object is "failed" or "error" since setting a low timeout might mislead us if the objects are still being created. Thoughts?